lib/HLSL/HLOperationLower.cpp

```diff
@@ -9465,25 +9463,26 @@ void EmitGetNodeRecordPtrAndUpdateUsers(HLOperationLowerHelper &helper,
   Value *opArg = nullptr;
   Value *Handle = CI->getArgOperand(HLOperandIndex::kHandleOpIdx);
   opArg = Builder.getInt32((unsigned)DXIL::OpCode::GetNodeRecordPtr);
-  StructType *origRecordUDT =
-      cast<StructType>(cast<PointerType>(CI->getType())->getElementType());
-  Type *getNodeRecordPtrRT = origRecordUDT;
-  // Translate node record type here
-  auto findIt = helper.loweredTypes.find(origRecordUDT);
-  if (findIt != helper.loweredTypes.end()) {
-    getNodeRecordPtrRT = findIt->second;
-  } else {
-    getNodeRecordPtrRT = GetLoweredUDT(origRecordUDT, &helper.dxilTypeSys);
-    if (origRecordUDT != getNodeRecordPtrRT)
-      helper.loweredTypes[origRecordUDT] = getNodeRecordPtrRT;
+  PointerType *pTy = cast<PointerType>(CI->getType());
+  if (StructType *origRecordUDT = dyn_cast<StructType>(pTy->getElementType())) {
```

Comment:
What will happen if the element type is not struct type?

Reply:
Should not be possible with earlier record template checks, so we should probably keep the `cast<>` we had before.  I'm not sure why this change was made here.

Reply:
This was a remnant of a change I needed to make for some reason that can be reverted now.